numpy/lib/_function_base_impl.py

```diff
@@ -3827,7 +3827,7 @@ def sinc(x):
 
     """
     x = np.asanyarray(x)
-    y = pi * where(x == 0, 1.0e-20, x)
+    y = pi * where(x, x, np.finfo(np.float64).smallest_normal)
```

Comment:
This results in casting all floating point inputs to float64; currently `float16` inputs result in `float16` outputs.

I think we should first cast `x` to float64 if it is integer typed, and then use `np.finfo(x.dtype)`

Reply:
Ah, oops, got it.

Reply:
> This would also fix the bug for np.sinc(float16(0)) in the current release!

what is the bug?